x-pack/plugins/lens/public/indexpattern_datasource/utils.ts

```diff
@@ -39,7 +39,8 @@ export function isDraggedField(fieldCandidate: unknown): fieldCandidate is Dragg
     typeof fieldCandidate === 'object' &&
     fieldCandidate !== null &&
     'field' in fieldCandidate &&
-    'indexPatternId' in fieldCandidate
+    'indexPatternId' in fieldCandidate &&
+    'id' in fieldCandidate
```

Comment:
nitpick, not a big deal, but moving to this format will be easier to extend in the future:

```js
['id', 'field', 'indexPatternId'].every( prop => prop in fieldCandidate)
```